### ISSUE TITLE
Consider adding the ability for user to change their own password

### ISSUE THREAD

**Issue by korydraughn** [irods/irods_client_http_api#59]
iCommands provide `iadmin` which allows rodsadmins to change any user's password.
iCommands also provide `ipasswd` which allows non-rodsadmins to change their own password.

Should the HTTP API provide an op that allows a user to change their own password without needing a rodsadmin?

How does this play into password reset UIs built on top of the HTTP API?

**Comment by trel** [comment]
And how does this play alongside OIDC, LDAP, where passwords are not stored within iRODS at all?   Do we need to have a per-call, per-user setting that says this is not-allowed?   Or discouraged?

**Comment by korydraughn** [comment]
IMO, this is specifically a native authentication thing. I think OIDC, etc are out of scope for this because they are their own thing with facilities for managing credentials.

Keep in mind I've never managed OIDC, LDAP, etc, so my knowledge of that space is very limited.

**Comment by trel** [comment]
Yes, if we can scope this to native, then I agree we should be able to open/access/manipulate this functionality via HTTP.

My question is whether the server knows that this particular user should not be allowed to set a password *because* it should only get accessed via LDAP/OIDC/etc...

**Comment by korydraughn** [comment]
Hmm. Have to think about that. Perhaps it's a config file thing?

At the moment, the config file is defined so that multiple auth schemes can be enabled. With that, the HTTP API relies on the Authorization HTTP header to determine which auth scheme logic should run.

Not sure LDAP is mapped to an authorization header value. We can always invent names, but if there's one that is standardized, I'm sure that's what we'd want to try and use.
